laplace/utils/swag.py

```diff
@@ -0,0 +1,81 @@
+from copy import deepcopy
+
+import torch
+from torch.nn.utils import parameters_to_vector
+
+
+def param_vector(model):
+    return parameters_to_vector(model.parameters()).detach()
+
+
+def fit_diagonal_swag(model, train_loader, criterion, n_snapshots_total=40, snapshot_freq=1, lr=0.01, momentum=0.9, weight_decay=3e-4, min_var=1e-30):
```

Comment:
Just out of interest: the mean is just discarded and not useful for the final LA or could it help to get a better mean following the idea of standard SWA? The function name indicates it does the full `swag` (mean + var) currently. If you just need the variance, maybe that can be indicated in the function name like `fit_diagonal_swag_var`. But it's not necessary, just a suggestion.

Reply:
Good point! iirc I have tried using the SWA mean with Laplace, which indeed improved performance. But I haven't done extensive experiments with it as I thought it's not as principled (although perhaps one could view the SWA mean as some kind of approximation to the MAP?), but it would be interesting to revisit the idea (and perhaps even add this as a feature if it works well). The only downside I see is that it requires us to update the BatchNorm parameters of the model, which is one of the main reasons why SWA(G) is so slow.

I've changed the name to `fit_diagonal_swag_var` for clarity, thanks!